dags/operators/moz_dataproc_operator.py

```diff
@@ -0,0 +1,451 @@
+        self.init_actions_uris = init_actions_uris
+        self.init_action_timeout = init_action_timeout
+        self.metadata = metadata
+        self.custom_image = custom_image
+        self.image_version = image_version
+        self.properties = properties or dict()
+        self.master_machine_type = master_machine_type
+        self.master_disk_type = master_disk_type
+        self.master_disk_size = master_disk_size
+        self.autoscaling_policy = autoscaling_policy
+        self.worker_machine_type = worker_machine_type
+        self.worker_disk_type = worker_disk_type
+        self.worker_disk_size = worker_disk_size
+        self.labels = labels
+        self.zone = zone
+        self.network_uri = network_uri
+        self.subnetwork_uri = subnetwork_uri
+        self.internal_ip_only = internal_ip_only
+        self.tags = tags
+        self.service_account = service_account
+        self.service_account_scopes = service_account_scopes
+        self.idle_delete_ttl = idle_delete_ttl
+        self.auto_delete_time = auto_delete_time
+        self.auto_delete_ttl = auto_delete_ttl
+        self.customer_managed_key = customer_managed_key
+        self.single_node = num_workers == 0
+
+        assert not (self.custom_image and self.image_version), \
+            "custom_image and image_version can't be both set"
+
+        assert (
+            not self.single_node or (
+                self.single_node and self.num_preemptible_workers == 0
+            )
+        ), "num_workers == 0 means single node mode - no preemptibles allowed"
+
+    def _get_init_action_timeout(self):
+        match = re.match(r"^(\d+)(s|m)$", self.init_action_timeout)
+        if match:
+            if match.group(2) == "s":
+                return self.init_action_timeout
+            elif match.group(2) == "m":
+                val = float(match.group(1))
+                return "{}s".format(timedelta(minutes=val).seconds)
+
+        raise AirflowException(
+            "DataprocClusterCreateOperator init_action_timeout"
+            " should be expressed in minutes or seconds. i.e. 10m, 30s")
+
+    def _build_gce_cluster_config(self, cluster_data):
+        """
+        We are going to modify this method to install jupyter/anaconda by default, because the original
+        operator doesnt support it.
+
+        cluster_data['config']['softwareConfig']['optionalComponents'] = ['ANACONDA','JUPYTER']
+
+        We also need to add alpha feature enable component gateway
+        cluster_data['config']['endpointConfig']['enableHttpPortAccess'] = True
+        note this is only available via the v1beta2 rest api, which the DataprocOperationBaseOperator
+        uses in master branch, and not in 1.10.2
+
+        """
+
+        # Fetch current nested dict and add nested keys
+        cluster_config_new = cluster_data['config']
+        cluster_config_new.update({'softwareConfig' : {'optionalComponents': ['ANACONDA','JUPYTER']}})
```

Comment:
There was some talk about making this configurable eventually. It's also a beta feature that appears to be unstable. which is unfortunate, but hopefully removing JUPYTER is a good start.